haskell/private/actions/link.bzl

```diff
@@ -252,19 +252,21 @@ def link_library_static(hs, cc, dep_info, objects_dir, my_pkg_id, with_profiling
               hs.tools_runfiles.ar + hs.extra_binaries)
 
     if hs.toolchain.is_darwin:
-        # ar is deprecated on Darwin, so we have to use libtool instead.
+        # On Darwin, ar doesn't support params files.
         args.add([
-            "-static",
-            "-o",
-            static_library.path,
-            "-filelist",
-            objects_dir_manifest,
+            static_library,
+            objects_dir_manifest.path,
         ])
-        hs.actions.run(
+
+        # TODO Get ar location from the CC toolchain. This is
```

Comment:
We still have an override making `hs.tools.ar` point to `/usr/bin/ar` on Mac:
https://github.com/tweag/rules_haskell/blob/7feaa397825d179045eb676c6bf1779f073d4ed7/haskell/toolchain.bzl#L145

I suggest using `hs.tools.ar` here, as we do on l.251.

Reply:
Good point.